doc-bridgepoint/notes/9110_Research_bp_profiling_tool/9110_Research_bp_profiling_tool.int.md

```diff
@@ -0,0 +1,207 @@
+
+Filtering is supported. You can compact filter classes of type (this is only going to show the first invocation).
+
+Steps to run are as follows:
+
+- Launch YourKit
+- Launch BridgePoint and choose a non-bridgepoint perspective (required due to a crash when profiling from workspace location choosing)
+- Restart BridgePoint
+- In YourKit click the underlined bridge point item under Monitor Lock Applications
+- In the toolbar choose Settings > Filters… to configure filters (if filtering is required)
+- Hover the Start CPU Profiling button at the top and choose the Tracing option
+- Click the Start CPU Profiling button again
+- Start loading BridgePoint by opening the xtUML Perspective
+- Once the tool is ready for user interaction (fully loaded) click the Stop CPU button
+- In the window opened examine the CPU runtimes collected
+- Note those that take the most time
+
+This one also supports snapshots and comparisons.  Again good for regression issues.
+
+Conclusion: This tool can be configured heavily but is not as intuitive as JProfiler.  The same data is provided as in JProfiler but the UI is not as elegant.
+
+Suggestion:
+
+Use JProfiler.  This profiler provides more data than jvisualvm, but either would work.  JProfiler is preferred just from a UI/intuitive standpoint over YourKit.  Easy to use quickly and all the same data is present.
+
+After determining this suggestion this website was found.  The suggestion is in agreement:
+
+https://blog.oio.de/2014/03/07/java-profilers-a-short-comparison-between-jprofiler-yourkit-and-javas-visualvm/
+
+Procedures
+----------
+
+JVM and Memory
+
+For all but memory leaks see:
+See: https://docs.oracle.com/cd/E26576_01/doc.312/e24936/tuning-java.htm#GSPTG00006
+
+- Look into tuning JVM, params like -Xms, 
+- Look into tuning GC, is it running too much?
+- Look for memory leaks, profile BridgePoint during the life and look through the collected memory data.  Perform deletions and watch the memory size assuring garbage collection occurs.  Also perform operations that close views/editors where temporary data should be garbage collected.
+- Look into UseParallelGC (garbage collects with multiprocessors)
+
+Application
+
+- Startup
+
+  • Launch profiler   
+  • Launch BridgePoint making sure a workspace chooser dialog is shown   
+  • Connect profiler to BridgePoint process   
+  • Begin profiling   
+  • Once BridgePoint is ready for user interaction pause profiling   
+  • Go to Investigate profiling data step   
+
+- After startup
+
+  • Launch profiler   
+  • Launch BridgePoint   
+  • Connect profiler to BridgePoint process   
+  • Right before UI interaction begin profiling (example moving a shape)   
+  • Once the UI interaction is complete pause profiling   
+  • Go to Investigate profiling data step   
+  
+- Regression testing
+
+  • For any of the above profiling procedures after running them save a snapshot
```

Comment:
If a snapshot must be saved, put the step to do so in the procedure above. Also, we need a place to keep these snapshots. Where we put them we depend on their size. If they are large we will use s3, if they are not large we can use a standard spot we create in github. This procedure needs to define this location